xapian-applications/omega/postprocess

```diff
@@ -0,0 +1,262 @@
+        test_clicks.flush()
+
+        test_final = tempfile.NamedTemporaryFile(delete=True)
+
+        generate_combined_log(test_search.name, test_clicks.name, test_final.name)
+
+        test_final.flush()
+
+        # Test entries in final log are correct.
+        with open(test_final.name, 'r') as final_f:
+            reader = csv.reader(final_f)
+
+            # Skip header row.
+            next(final_f)
+
+            for i, row in enumerate(reader):
+                if i == 0:
+                    assert row == ['821f03288846297c2cf43c34766a38f7',
+                                  'book', '45,54', '0', '45:0,54:2'], "Incorrect entry in final.log"
+                if i == 1:
+                    assert row == ['098f6bcd4621d373cade4e832627b4f6',
+                                  'test', '35,47', '0', '35:1,47:0'], "Incorrect entry in final.log"
+
+        test_query = tempfile.NamedTemporaryFile(delete=True)
+
+        generate_query_file(test_final.name, test_query.name)
+
+        test_query.flush()
+
+        # Test entries in query file are correct.
+        with open(test_query.name, 'r') as query_f:
+            reader = csv.reader(query_f)
+
+            for i, row in enumerate(reader):
+                if i == 0:
+                    assert row == ['821f03288846297c2cf43c34766a38f7','book'], "Incorrect entry in query.txt"
+                if i == 1:
+                    assert row == ['098f6bcd4621d373cade4e832627b4f6','test'], "Incorrect entry in query.txt"
+    finally:
+        # Close all files to safely delete them.
+        try:
+            test_search.close()
+        except IOError as e:
+            print("Error: %s - %s." % (e.filename, e.strerror))
+        try:
+            test_clicks.close()
+        except IOError as e:
+            print("Error: %s - %s." % (e.filename, e.strerror))
+        try:
+            test_final.close()
+        except IOError as e:
+            print("Error: %s - %s." % (e.filename, e.strerror))
+        try:
+            test_query.close()
+        except IOError as e:
+            print("Error: %s - %s." % (e.filename, e.strerror))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='''Postprocess click data files.
+
+This script generates the final clickstream log file from input search and
+click log files and creates query file that can be used by the Xapian Letor
+module for generating its training files.''',
+formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument("search_log", type=str, help="Path to the search.log file.")
+    parser.add_argument("clicks_log", type=str, help="Path to the clicks.log file.")
+    parser.add_argument("final_log", type=str, help="Path to save final.log file.")
+    parser.add_argument("query_file", type=str, help="Path to save query.txt file.")
+    parser.add_argument("--test", help="Run tests for this script.", action='store_true')
```

Comment:
Given `--test` is internal only, I think it's fine to pass through dummy args in this case. That's better than having to write more code to cope with the non-test case and ensure all the args have been passed correctly.